src/hooks/usePollDatasetStatus.js

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState, useRef } from 'react'
+import { useDatasetManager } from 'hooks/useDatasetManager'
+
+export const usePollDatasetStatus = () => {
+  const {
+    datasetAccessions,
+    setDatasetAccessions,
+    processingDatasets,
+    setProcessingDatasets,
+    getDataset
+  } = useDatasetManager()
+  const [polledDatasetId, setPolledDatasetId] = useState(null)
+  const [polledDatasetState, setPolledDatasetState] = useState(false)
+  const timerRef = useRef(null)
+  const isProcessingDataset = polledDatasetState?.is_processing
```

Comment:
```suggestion
  const [isProcessing, setIsProcessing] = useState(false)
```
If this is the only output we care about let's keep this a state and have `refreshProcessingDataset` handle setting this to true or false accordingly.

Reply:
I've removed this local variable and added a new state `isProcessingDataset` (boolean), see at a5bf82650a3adc2e4925149acbcb0b3b4820ee21.